### ISSUE TITLE
react/prop-types does not detect "is missing in props validation" if some of the props are passed to other components.

### ISSUE THREAD

**Issue by olegserov** [yannickcr/eslint-plugin-react#2091]
If some props are being passed to some other components, it fails to detect `error  'varname' is missing in props validation         react/prop-types`

.eslintrc: 
```
{
  "extends": [
    "standard",
    "standard-react"
  ],
  "parser": "babel-eslint"
}

```
Packages:
```
 "eslint": "^5.10.0",
"eslint-plugin-react": "^7.11.1",
 "eslint-plugin-standard": "^4.0.0",
```

Detects error:
```javascript
import React from 'react'

export default class Test extends React.Component {
  static propTypes = {}

  render () {
    const {
      varname,
      ...otherProps
    } = this.props

    return <div>
      {varname}
    </div>
  }
}
```

Does not detect the error
```javascript
import React from 'react'

export default class Test extends React.Component {
  static propTypes = {}

  render () {
    const {
      varname,
      ...otherProps
    } = this.props

    return <div {...otherProps}>
      {varname}
    </div>
  }
}

```

**Comment by ljharb** [comment]
I believe this is fixed in latest maser, but not yet released.

**Comment by hugomallet** [comment]
It works with current version (7.12.4) of eslint-plugin-react
